targets/psa_switch/externs/psa_hash.cpp

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2021 SYRMIA LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Dusan Krdzic (dusan.krdzic@syrmia.com)
+ *
+ */
+
+
+#include "psa_hash.h"
+
+namespace {
+
+void
+build_buffer(const std::vector<bm::Field> &fields, bm::ByteContainer &buf) {
```

Comment:
according to the coding style guide for this repo, we use pointers for out parameters (parameters that are modified by the function), so it should be `bm::ByteContainer *buf`. However, it seems that here `build-buffer` could return the `bm::ByteContainer` object directly, benefiting from NRVO? `bm::ByteContainer build_buffer(const std::vector<bm::Field> &fields)`?

Reply:
Ok. I will do in that way.